src/framework/execution/__tests__/dependencyGraph.test.ts

```diff
@@ -649,4 +685,164 @@ describe('executeStepDependencyGraph', () => {
     expect(spyC).toHaveBeenCalledBefore(spyE);
     expect(spyD).toHaveBeenCalledBefore(spyE);
   });
+
+  test('does not execute step if marked ', async () => {
+    const spyA = jest.fn();
+
+    /**
+     *     e
+     *       \
+     * a - c - d
+     *       /
+     *     b
+     *
+     * In this situation, 'a', 'e', and 'b' are leaf nodes
+     * 'c' depends on 'a',
+     * 'd' depends on 'e', 'c', and 'b'
+     */
+    const steps: IntegrationStep[] = [
+      {
+        id: 'a',
+        name: 'a',
+        types: [],
+        executionHandler: spyA,
+      },
+    ];
+
+    const graph = buildStepDependencyGraph(steps);
+    const stepStartStates = getDefaultStepStartStates(steps);
+    stepStartStates['a'] = {
+      disabled: true,
+    };
+
+    const result = await executeStepDependencyGraph(
+      executionContext,
+      graph,
+      stepStartStates,
+    );
+
+    expect(spyA).toHaveBeenCalledTimes(0);
+
+    expect(result).toEqual([
+      {
+        id: 'a',
+        name: 'a',
+        types: [],
+        status: IntegrationStepResultStatus.NOT_EXECUTED,
```

Comment:
I wonder if we need a more specific status of `DISABLED`, to clarify the reason.

Reply:
Ah yeah, `DISABLED` would be more clear match up with the `disabled` field in the start states. Will change.